File: Semester4/Systems for Design and Implementation/mpp-proiect-csharp-raulfrandes/model/validator/RefereeValidator.cs
namespace model.validator;

public class RefereeValidator: IValidator<Referee>
{
    public void Validate(Referee entity)
    {
        string errorMess = "";
        if (string.IsNullOrEmpty(entity.Name))
        {
            errorMess += "Invalid name - cannot be null\n";
        }
        if (string.IsNullOrEmpty(entity.Username))
        {
            errorMess += "Invalid name - cannot be null\n";
        }
        if (string.IsNullOrEmpty(entity.Password))
        {
            errorMess += "Invalid password - cannot be null\n";
        }
        if (entity.Password.Length < 4)
        {
            errorMess += "Invalid password - must have at least 4 characters";
        }
        if (!string.IsNullOrEmpty(errorMess))
        {
            throw new ValidatorException(errorMess);
        }
    }
}

File: Semester4/Systems for Design and Implementation/mpp-proiect-java-raulfrandes/TriathlonModel/src/main/java/triathlon/model/validator/RefereeValidator.java
package triathlon.model.validator;


import triathlon.model.Referee;

public class RefereeValidator implements Validator<Referee> {
    @Override
    public void validate(Referee entity) throws ValidatorException {
        String errorMess = "";
        if (entity.getName().isEmpty()) {
            errorMess += "Invalid name - cannot be null\n";
        }
        if (entity.getUsername().isEmpty()) {
            errorMess += "Invalid name - cannot be null\n";
        }
        if (entity.getPassword().isEmpty()) {
            errorMess += "Invalid password - cannot be null\n";
        }
        if (entity.getPassword().length() < 4) {
            errorMess += "Invalid password - must have at least 4 characters";
        }
        if (!errorMess.isEmpty()){
            throw new ValidatorException(errorMess);
        }
    }
}
